src/rulesets/LinearAlgebra/blas.jl

```diff
@@ -17,15 +17,13 @@ rrule(::typeof(BLAS.dot), x, y) = rrule(dot, x, y)
 
 function rrule(::typeof(BLAS.dot), n, X, incx, Y, incy)
     Ω = BLAS.dot(n, X, incx, Y, incy)
+    function blas_dot_pullback(::Zero)
+        return (NO_FIELDS, DoesNotExist(), Zero(), DoesNotExist(), Zero(), DoesNotExist())
+    end
     function blas_dot_pullback(ΔΩ)
-        if ΔΩ isa Zero
```

Comment:
why is this type unstable?
I mean we don't need this branch because its always applicable to do this.
So its not something to include in the rule.
but i want to understand.

Reply:
Oh, actually, it looks like `extern` caused the type-instability in the type of `InplaceableThunk`. That seems like it should not happen? But I can revert my change and accept the one you provided.

Reply:
Reassigning a variable to contain another type causes type instablilties.
I think it shouldn't (I think the compiler should declare a second slot if that can happen and swap to using that but i guess it is hard to workout in adance, possibly impossible).
This basically makes the slow dynamically types
Often the compilere does compile it away.
But not always.